File: codec/src/main/java/top/catnemo/codec/util/Logger.java
package top.catnemo.codec.util;

import android.util.Log;
import top.catnemo.codec.BuildConfig;

/**
 *
 * @author matrixJoy
 * @description
 * @date 2019-11-12   10:57 
 */
public class Logger {

    private static boolean isDebug = BuildConfig.DEBUG;

    public static void v(String tag, String msg) {
        if (isDebug) {
            Log.v(tag, msg);
        }
    }

    public static void d(String tag, String msg) {
        if (isDebug) {
            Log.d(tag, msg);
        }
    }


    public static void w(String tag, String msg) {
        if (isDebug) {
            Log.w(tag, msg);
        }
    }

    public static void e(String tag, String msg) {
        if (isDebug) {
            Log.e(tag, msg);
        }
    }

    public static void e(String tag, String msg, Throwable throwable) {
        if (isDebug) {
            Log.e(tag, msg, throwable);
        }
    }

}

File: zfilter/src/main/java/com/catnemo/zfilter/util/Logger.kt
package com.catnemo.zfilter.util

import android.util.Log
import com.catnemo.zfilter.BuildConfig

/**
 *
 * @author matrixJoy
 * @description
 * @date 2019-11-12   10:57
 */
object Logger {

    private val isDebug = BuildConfig.DEBUG

    fun v(tag: String, msg: String) {
        if (isDebug) {
            Log.v(tag, msg)
        }
    }

    fun d(tag: String, msg: String) {
        if (isDebug) {
            Log.d(tag, msg)
        }
    }


    fun w(tag: String, msg: String) {
        if (isDebug) {
            Log.w(tag, msg)
        }
    }

    fun e(tag: String, msg: String) {
        if (isDebug) {
            Log.e(tag, msg)
        }
    }

    fun e(tag: String, msg: String, throwable: Throwable) {
        if (isDebug) {
            Log.e(tag, msg, throwable)
        }
    }

}
